### PR TITLE
Removing deprecated samples pages and redirecting to new samples repo

### PR DESCRIPTION
**Pull request by sfshaza2**
Tested here:  https://sz-flutter.firebaseapp.com/

For example:  
- https://sz-flutter.firebaseapp.com/docs/catalog/samples
- https://sz-flutter.firebaseapp.com/docs/catalog/samples/expansion-tile-sample

Fixes https://github.com/flutter/website/issues/4591
Fixes https://github.com/flutter/website/issues/4583

Old deprecated samples pages were being found by search engines. These URLs now redirect to our current samples.

### CODE REVIEW

File: firebase.json [hunk 1]
```diff
@@ -49,12 +49,6 @@
       { "source": "/assets-and-images", "destination": "/docs/development/ui/assets-and-images", "type": 301 },
       { "source": "/bootstrap-into-dart", "destination": "/docs/resources/bootstrap-into-dart", "type": 301 },
       { "source": "/bug-reports", "destination": "/docs/resources/bug-reports", "type": 301 },
-      { "source": "/catalog/samples", "destination": "/docs/catalog/samples", "type": 301 },
-      { "source": "/catalog/samples/tabbed-app-bar", "destination": "/docs/cookbook/design/tabs", "type": 301 },
-      { "source": "/catalog/samples/:page*_index", "destination": "/docs/catalog/samples/:page*", "type": 301 },
-      { "source": "/catalog/samples/AppBar_index", "destination": "/docs/catalog/samples/AppBar", "type": 301 },
-      { "source": "/catalog/samples/TabBar_index", "destination": "/docs/catalog/samples/TabBar", "type": 301 },
-      { "source": "/catalog/samples/:rest*", "destination": "/docs/catalog/samples/:rest*", "type": 301 },
```
Comment (kwalrath) [inline firebase.json:57]: Do we want to keep a couple of /catalog/samples redirects (similar to the /docs/catalog/samples redirects) to forestall 404s? Or do you want to find out whether people are still using these URLs, even if it causes 404s?

**Review by kwalrath — APPROVED** [review]
LGTM, although I wonder if we might not want to keep a couple of /catalog redirects.

File: firebase.json (hunk 1 above)
Comment (sfshaza2) [reply firebase.json:57]: I'd rather wait and see.  :)

Comment (kwalrath) [reply firebase.json:57]: More work for me. I see. 😄

**Pull request merged**